CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.17.0)
+cmake_minimum_required(VERSION 3.16.0)
```

Comment:
Is there a specific reason to downgrade the cmake version requirement here?

Reply:
The pragmatic reason was that I only had version 3.16 of cmake available in the base image (Ubuntu focal, providing [v3.16.3](https://packages.ubuntu.com/focal/cmake)) that I used to provision the virtual machine via Vagrant. I used a VM to do the build before you fixed the issues with more recent compiler and libc++ versions. 

Anyway, I did not notice any problems with the v3.16.0 requirement. Was there any specific reason for you to enforce v3.17.0, or was this just the version you had at hand?

Reply:
Sounds totally reasonable to me. It was just the version I had at hand and I just wanted to know why downgrading the requirement makes sense. :)